Packs/RiskIQDigitalFootprint/Integrations/RiskIQDigitalFootprint/RiskIQDigitalFootprint.py

```diff
@@ -0,0 +1,2451 @@
+        'Enterprise Asset': resp.get('enterprise', None),
+        'Priority': resp.get('priority', ''),
+        'Organization(s)': get_comma_separated_values(resp.get('organizations', []), 'name'),
+        'Tag(s)': get_comma_separated_values(resp.get('tags', []), 'name'),
+        'Brand(s)': get_comma_separated_values(resp.get('brands', []), 'name')
+    }
+
+
+def get_asset_outputs(resp: Dict[str, Any]) -> Tuple[str, list, Dict[str, Any]]:
+    """
+    Generates all outputs for df-get-asset command
+
+    :param resp:  API Response
+    :return: Human Readable, Standard Context, Custom Context
+    """
+    hr = ''
+    custom_ec = {}
+    standard_ec: list = []
+    asset_type_hr_map = {
+        'DOMAIN': get_asset_domain_hr,
+        'HOST': get_asset_host_hr,
+        'IP_ADDRESS': get_asset_ip_address_hr,
+        'IP_BLOCK': get_asset_ip_block_hr,
+        'AS': get_asset_as_hr,
+        'PAGE': get_asset_page_hr,
+        'SSL_CERT': get_asset_ssl_hr,
+        'CONTACT': get_asset_contact_hr
+    }
+    asset_type_standard_context_map = {
+        'DOMAIN': get_standard_context_domain_for_get_asset,
+        'HOST': get_standard_context_host_for_get_asset,
+        'IP_ADDRESS': get_standard_context_ip_for_get_asset,
+        'PAGE': get_standard_context_url_for_get_asset
+    }
+    if resp:
+        asset_type = resp.get('type', '')
+        asset_details = resp.get('asset', {})
+
+        custom_ec = get_asset_custom_context(resp)
+        hr += '### ASSET DETAILS\n'
+        asset_basic_hr = get_asset_basic_hr(resp)
+        hr += tableToMarkdown('Basic Details', asset_basic_hr,
+                              ['Name', 'Type', 'Inventory Status', 'UUID', 'First Seen (GMT)', 'Last Seen (GMT)',
+                               'Added To Inventory (GMT)', 'Last Updated (GMT)', 'Confidence', 'Enterprise Asset',
+                               'Priority', 'Organization(s)', 'Tag(s)', 'Brand(s)'],
+                              removeNull=True)
+
+        if asset_type in asset_type_hr_map:
+            hr += asset_type_hr_map[asset_type](asset_details)
+        if asset_type in asset_type_standard_context_map:
+            standard_ec = asset_type_standard_context_map[asset_type](resp)
+        custom_ec = createContext(custom_ec, removeNull=True)
+
+    return hr, standard_ec, custom_ec
+
+
+def get_add_and_update_assets_params(args: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Fetch and validate parameters to be passed for df-add-assets or df-update-assets command
+    and prepare params dictionary.
+
+    :param args: args passed by the user
+    :return: params dictionary
+    """
+    params: Dict[str, Any] = {}
+
+    if args.get('fail_on_error'):
+        fail_on_error = args.get('fail_on_error')
+        if fail_on_error not in ['true', 'false']:
+            raise ValueError(MESSAGES['INVALID_BOOLEAN_VALUE_ERROR'].format('fail_on_error'))
+        params['failOnError'] = argToBoolean(fail_on_error)
+    return params
```

Comment:
This strikes me as overkill. I think you could do this in a simpler way.

Reply:
@avidan-H Sure we will try to reduce this.